From 989100e5a86e5337f5f5dae0c00fb7ffc741fba2 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 7 Jul 2023 09:39:50 +0530
Subject: [PATCH 0450/1141] arm64: dts: stm32: add rpmsg i2c stm32g0-typec
 device for stm32mp257f-ev1 boards

Add the support of the stm32go typeC device on the rpmsg I2C adapter.
Instead of communicating with an external G0 processor, it communicates
with the coprocessor relying on a virtual i2c. G0  phuysical interrupts
are replaced by a rpmsg virtual interrupts controller through the
irq-rpmsg service.
notice that this impose that the rpmsg framework treat message under
interrupt context. this solution should be replaced by virtio-gpio and
virtio-i2c solutions.

Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Change-Id: Ide60b1de4b1a2b2bbfca859d108caad7cebfbe4d
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -450,6 +450,13 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 		status = "okay";
+
+		typec@35 {
+			compatible = "st,stm32mp25-typec";
+			reg = <0x35>;
+			interrupts-extended = <&intc_rpmsg 0>;
+			status = "okay";
+		};
 	};
 };
 
